publish.go

```diff
@@ -164,3 +168,36 @@ type BatchItemsError []BatchItemResponse
 func (err BatchItemsError) Error() string {
 	return "one or many events may have not been published"
 }
+
+// Format implements fmt.Formatter for BatchItemsError
+func (err BatchItemsError) Format(s fmt.State, verb rune) {
+	switch verb {
+	case 'v':
+		if s.Flag('+') {
+			var messages []string
+			for k, v := range err {
+				messages = append(messages, fmt.Sprintf("[%d]: %+v", k, v))
+			}
+			sort.Strings(messages)
+
+			builder := bytes.NewBuffer(make([]byte, 0))
+			switch len(err) {
+			case 0:
+				builder.WriteString("an error occurred while publishing event")
```

Comment:
I think it would make sense to be more precise about this weird case when we have some errors but also not.
Perhaps `an unknown error occurred while publishing event` could do the trick, maybe we could be even more explicit.
@stoewer ?